pilot/pkg/bootstrap/server.go

```diff
@@ -1125,4 +1125,6 @@ func (s *Server) initMeshHandlers() {
 			Reason: []model.TriggerReason{model.GlobalUpdate},
 		})
 	})
+
+	s.environment.InitClusterLocalHosts()
```

Comment:
This needs to be moved to the `AddMeshHandler` function above ... before we do anything with the `XDSServer`.